Fixes a few bugs in the R-group decomposition code

ptosco opened the pull request:
* Even when setting `RGroupLabels` other than to `AutoDetect`, the MCS would still insist to match R-group labels to dummy atoms causing problems (detailed description of the bug here: https://gist.github.com/ptosco/7b6603f1bf665262c1d6022a2c2fccfa and in the new test cases I have added)
* R groups with no substituents in any of the `Add`ed molecules were listed by `GetRGroupsAsColumns`
* Assignment of R-group labels had bugs connected to how `maxLabel` was being set: some indices were missed (e.g., you would see `R1` and `R3` but no `R2`) and things went seriously wrong when using `DummyLabels`.
* `MDLRGroupLabels` and `DummyLabels` were not exposed to Python
* I have added `RGroupCoreAlignment.None_` (while keeping `RGroupCoreAlignment.None` for backwards compatibility) to address the issue reported by @rvianello in #3346)

bp-kelley commented:
@ptosco you beat me to it, I was a working on some similar issues.

The osx builds look real.  And perhaps real annoying.

There is another issue where the output rgroups aren't necessarily in order, i.e. R2...R4...R5 this was a regression from a previous build that apparently I forgot to put regression tests on, I'll work on it after we sort these.

bp-kelley commented on Code/GraphMol/RGroupDecomposition/Wrap/rdRGroupComposition.cpp:
```diff
@@ -200,25 +202,31 @@ struct rgroupdecomp_wrapper {
 
     python::enum_<RDKit::RGroupCoreAlignment>("RGroupCoreAlignment")
         .value("None", RDKit::None)
+        .value("None_", RDKit::None)
```
Let's just do a breaking change here and switch it to NoAlignment both ini C++ and in Python.

I never should have used None in the first place ;)

bp-kelley commented on Code/GraphMol/RGroupDecomposition/RGroupDecompParams.cpp:
```diff
@@ -53,6 +53,38 @@ unsigned int RGroupDecompositionParameters::autoGetLabels(const RWMol &core) {
   return autoLabels;
 }
 
+bool rgdAtomCompare(const MCSAtomCompareParameters &p, const ROMol &mol1,
```
If the user took the trouble to label R1 let's assume that's what they want as R1.

bp-kelley replied:
@ptosco @greglandrum One remaining issue here.

ptosco commented:
@greglandrum I fixed the Mac bug (it was a silly one, but it took me a while to set up the build environment on a macincloud instance - my own Mac has become too old to be useful :(. I also added some more tests to cover the `NoAlignment` case.
@bp-kelley this is good to go for me - it would be great if you had a few minutes to sign it off for merging.
I think it is definitely worth fixing these RGD bugs before release.

bp-kelley commented:
@ptosco  there was one remaining issue but I may have convinced myself not to worry abut it for now.
